File: languages/FSML/Python/FsmlAcceptor.py
from antlr4 import *
from FsmlExceptions import FsmlException
from antlr4.error.Errors import ParseCancellationException
from FsmlLexer import FsmlLexer
from FsmlParser import FsmlParser
import sys

def parse(path):
    input = FileStream(path)
    lexer = FsmlLexer(input)
    tokens = CommonTokenStream(lexer)
    parser = FsmlParser(tokens)
    parser.fsm()
    return parser.model.fsm

try:
    fsm = parse(sys.argv[1])
except ParseCancellationException:
    sys.exit(1)

File: languages/FSML/Java/org/softlang/fsml/FsmlAcceptor.java
package org.softlang.fsml;

import org.antlr.v4.runtime.ANTLRFileStream;
import org.antlr.v4.runtime.CommonTokenStream; 
import java.io.IOException;

public class FsmlAcceptor {
	public static void main(String[] args)
			throws IOException {
		FsmlParser parser =
		  new FsmlParser(
		    new CommonTokenStream(
		      new FsmlLexer(
		    	new ANTLRFileStream(args[0]))));
		parser.fsm();
		System.exit(parser.getNumberOfSyntaxErrors());
	}
}
